examples/all-clusters-app/all-clusters-common/include/oven-operational-state-delegate.h

```diff
@@ -0,0 +1,116 @@
+class OvenCavityOperationalStateDelegate : public OperationalState::Delegate
+{
+private:
+    const Clusters::OperationalState::GenericOperationalState mOperationalStateList[8] = {
+        OperationalState::GenericOperationalState(to_underlying(OperationalState::OperationalStateEnum::kStopped)),
+        OperationalState::GenericOperationalState(to_underlying(OperationalState::OperationalStateEnum::kRunning)),
+        OperationalState::GenericOperationalState(to_underlying(OperationalState::OperationalStateEnum::kPaused)),
+        OperationalState::GenericOperationalState(to_underlying(OperationalState::OperationalStateEnum::kError)),
+    };
+
+    const Clusters::OperationalState::GenericOperationalPhase mOperationalPhaseList[1] = {
+        // Phase List is null
+        OperationalState::GenericOperationalPhase(DataModel::Nullable<CharSpan>()),
+    };
+
+public:
+    /**
+     * Get the countdown time. This attribute is not supported in our example app.
+     * @return Null.
+     */
+    DataModel::Nullable<uint32_t> GetCountdownTime() override { return {}; };
+
+    /**
+     * Fills in the provided GenericOperationalState with the state at index `index` if there is one,
+     * or returns CHIP_ERROR_NOT_FOUND if the index is out of range for the list of states.
+     * Note: This is used by the SDK to populate the operational state list attribute. If the contents of this list changes,
+     * the device SHALL call the Instance's ReportOperationalStateListChange method to report that this attribute has changed.
+     * @param index The index of the state, with 0 representing the first state.
+     * @param operationalState  The GenericOperationalState is filled.
+     */
+    CHIP_ERROR GetOperationalStateAtIndex(size_t index,
+                                          Clusters::OperationalState::GenericOperationalState & operationalState) override;
+
+    /**
+     * Fills in the provided GenericOperationalPhase with the phase at index `index` if there is one,
+     * or returns CHIP_ERROR_NOT_FOUND if the index is out of range for the list of phases.
+     * Note: This is used by the SDK to populate the phase list attribute. If the contents of this list changes, the
+     * device SHALL call the Instance's ReportPhaseListChange method to report that this attribute has changed.
+     * @param index The index of the phase, with 0 representing the first phase.
+     * @param operationalPhase  The GenericOperationalPhase is filled.
+     */
+    CHIP_ERROR GetOperationalPhaseAtIndex(size_t index,
+                                          Clusters::OperationalState::GenericOperationalPhase & operationalPhase) override;
+
+    // command callback
+    /**
+     * Handle Command Callback in application: Pause
+     * @param[out] get operational error after callback.
+     */
+    void HandlePauseStateCallback(Clusters::OperationalState::GenericOperationalError & err) override{
+        // This command in not supported.
```

Comment:
Shouldn't it error out, then?  Similar for the other ones below.

Reply:
errors now being set for this and similar cases.